lumicks/pylake/tests/test_channels.py

```diff
@@ -372,14 +372,55 @@ def test_downsampling_to():
     assert np.all(np.equal(s2a.timestamps, [0, 20, 40, 60, 80]))
     assert np.allclose(s2a.data, [2.5,  6.5, 10.5, 14.5, 18.5])
     s2b = s.downsampled_to(0.5e8, where='center')
-    assert np.all(np.equal(s2b.timestamps, [10, 30, 50, 70, 90]))
+    assert np.all(np.equal(s2b.timestamps, np.array([15/2, 55/2, 95/2, 135/2, 175/2], dtype=np.int)))
     assert np.allclose(s2b.data, [2.5,  6.5, 10.5, 14.5, 18.5])
 
     # to 33 ns step
     s3a = s.downsampled_to(0.3e8, where='left')
     assert np.all(np.equal(s3a.timestamps, [0, 33, 66]))
     assert np.allclose(s3a.data, [4.0, 11.0, 17.5])
     s3b = s.downsampled_to(0.3e8, where='center')
-    assert np.all(np.equal(s3b.timestamps, [16, 50, 83]))
+    assert np.all(np.equal(s3b.timestamps, [15, 50, 82]))
     assert np.allclose(s3b.data, [4.0, 11.0, 17.5])
 
+
+def test_downsampling_consistency():
+    d = np.arange(1, 24)
+    s = channel.Slice(channel.Continuous(d, 0, 10))
+
+    # Multiple of 5 should downsample to the same irrespective of the method
+    # Source frequency was 1e9 / 10 Hz. So we go to .2e8 Hz.
+    s1 = s.downsampled_to(.2e8)
+    s2 = s.downsampled_by(5)
```

Comment:
Maybe include a test with a not so well-behaved timestep (like 7ns)? I have a feeling this might also bring out the odd behavior mentioned in `test_downsampling_to()`

Reply:
I can add one with 7 ns source timestep. As long as the `downsampled_by` ratio is integer, everything should be fine in the new implementation. For non-integer fractions of the original samplerate you cannot get consistency by definition, only tradeoffs, as you're going into resampling territory.